tools/rosgraph/src/rosgraph/xmlrpc.py

```diff
@@ -76,9 +77,33 @@ def isstring(s):
     except NameError:
         return isinstance(s, str)
 
-class SilenceableXMLRPCRequestHandler(SimpleXMLRPCRequestHandler):
 
-    protocol_version = 'HTTP/1.1'
+def check_linux_kernel(major, minor):
+    """Small helper version to check that the current Linux kernel is a
+    newer or the same as the input.
+    """
+    release = platform.release().split('.')
+
+    platform_major = int(release[0])
+    platform_minor = int(release[1])
+
+    if platform_major > major:
+        return True
+    elif platform_major < major:
+        return False
+
+    if platform_minor > minor:
+        return True
+    elif platform_minor < minor:
+        return False
+
+    return True
+
+
+class SilenceableXMLRPCRequestHandler(SimpleXMLRPCRequestHandler):
+    # Linux kernels 4.15 and older encounter performance issues with HTTP/1.1
+    if platform.system() == 'Linux' and check_linux_kernel(4,16):
```

Comment:
tiny nitpick - i think you could have the `platform.system() == 'Linux'` inside the `check_linux_kernel` function, it seems appropriate to have that utility `return False` on non-Linux systems